mpqp/core/instruction/measurement/pauli_string.py

```diff
@@ -309,6 +313,98 @@ def from_matrix(cls, matrix: Matrix) -> PauliString:
                 PauliStringMonomial(0, [I for _ in range(num_qubits)])
             )
         return pauli_list
+    @classmethod
+    def _get_nb_qubits_cirq_pauli(
+        cls, pauli: Union[CirqPauliSum, CirqPauliString]
+    ):
+        from cirq.ops.pauli_string import PauliString as CirqPauliString
+        from cirq.ops.linear_combinations import PauliSum as CirqPauliSum
+
+        nb_qubits = 0
+        if isinstance(pauli, CirqPauliSum):
+            for term in pauli:
+                for qubit, _ in term.items():
+                    nb_qubits = max(nb_qubits, int(qubit.name.split("_")[1]) + 1)
+        elif isinstance(pauli, CirqPauliString):
+            for qubit in pauli.qubits:
+                nb_qubits = max(nb_qubits, int(qubit.name.split("_")[1]) + 1)
+        return nb_qubits
+    
+    @classmethod
+    def _from_cirq(
+        cls, pauli: Union[CirqPauliSum, CirqPauliString], nb_qubit: int = 0
+    ) -> PauliString:
+        from mpqp.measures import I, X, Y, Z
+        from mpqp.core.instruction.measurement.pauli_string import PauliStringMonomial
+        from cirq.ops.pauli_gates import X as Cirq_X, Y as Cirq_Y, Z as Cirq_Z
+        from cirq.ops.pauli_string import PauliString as CirqPauliString
+        from cirq.ops.linear_combinations import PauliSum as CirqPauliSum
+
+        def get_atom(op: Any):
+            if op == Cirq_X:
+                return X
+            elif op == Cirq_Y:
+                return Y
+            elif op == Cirq_Z:
+                return Z
+            else:
+                raise ValueError(f"Unsupported gate: {op}")
+
+        num_qubits = (
+            PauliString._get_nb_qubits_cirq_pauli(pauli) if nb_qubit == 0 else nb_qubit 
+        )
+        pauli_string = PauliString()
+
+        if isinstance(pauli, CirqPauliSum):
+            for term in pauli:
+                coef = term.coefficient.real
+                atoms = [I] * num_qubits
+                for qubit, op in term.items():
+                    nb_qubit = int(qubit.name.split("_")[1])
+                    atoms[nb_qubit] = get_atom(op)
+                pauli_string += PauliStringMonomial(coef, atoms)
+
+        elif isinstance(pauli, CirqPauliString):
+            coef = pauli.coefficient.real
+            atoms = [I] * num_qubits
+            for qubit, op in pauli.items():
+                nb_qubit = int(qubit.name.split("_")[1])
+                atoms[nb_qubit] = get_atom(op)
+            pauli_string += PauliStringMonomial(coef, atoms)
+
+        return pauli_string
+
+    @classmethod
+    def from_other_languages(
+        cls, pauli: Union[CirqPauliSum, CirqPauliString, List[CirqPauliString]], nb_qubits: int = 0
+    ) -> PauliString | List[PauliString]:
```

Comment:
why do we need to pass a list of PS ?

Reply:
we could also add PS from other SDKs but put them behind an `NotImplementedError`